### PR TITLE
PLT-24:Added example for vsphere cluster with DDNS & Static IP

### PR DESCRIPTION
**Pull request by SivaanandM**

### CODE REVIEW

File: examples/resources/spectrocloud_cluster_vsphere/providers.tf [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    }
+  }
+}
+
+variable "sc_host" {
+  description = "Spectro Cloud Endpoint"
+  default     = "api.spectrocloud.com"
+}
+
+variable "sc_username" {
```
Comment (nikchern) [inline examples/resources/spectrocloud_cluster_vsphere/providers.tf:15]: replace with key.

File: examples/resources/spectrocloud_cluster_vsphere/providers.tf [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  description = "Spectro Cloud Username"
+}
+
+variable "sc_password" {
```
Comment (nikchern) [inline examples/resources/spectrocloud_cluster_vsphere/providers.tf:19]: replace with key

**Review by nikchern — CHANGES_REQUESTED** [review]

File: examples/resources/spectrocloud_cluster_vsphere/resource.tf [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  name = "gm-pcg-wop-d"
+}
+
+
+resource "spectrocloud_cluster_vsphere" "cluster" {
+  name               = "vsphere-picard-2"
+  cluster_profile_id = data.spectrocloud_cluster_profile.vmware_profile.id
+  cloud_account_id   = data.spectrocloud_cloudaccount_vsphere.vmware_account.id
+
+  cluster_rbac_binding {
```
Comment (nikchern) [inline examples/resources/spectrocloud_cluster_vsphere/resource.tf:16]: Let's remove rbac, namespaces and other things which are not related to the example.

**Review by nikchern — CHANGES_REQUESTED** [review]

File: examples/resources/spectrocloud_cluster_vsphere/providers.tf (hunk 1 above)
Comment (SivaanandM) [reply examples/resources/spectrocloud_cluster_vsphere/providers.tf:15]: done

File: examples/resources/spectrocloud_cluster_vsphere/resource.tf (hunk 3 above)
Comment (SivaanandM) [reply examples/resources/spectrocloud_cluster_vsphere/resource.tf]: removed

File: examples/resources/spectrocloud_cluster_vsphere/terraform.templare.tfvars [hunk 4]
```diff
@@ -0,0 +1,30 @@
+# Spectro Cloud credentials
+sc_host         = "{enter Spectro Cloud API endpoint}" #e.g: api.spectrocloud.com (for SaaS)
+sc_username     = "{enter Spectro Cloud username}"     #e.g: user1@abc.com
+sc_password     = "{enter Spectro Cloud password}"     #e.g: supereSecure1!
```
Comment (nikchern) [inline examples/resources/spectrocloud_cluster_vsphere/terraform.templare.tfvars:4]: let's remove all password usage from examples.

File: examples/resources/spectrocloud_cluster_vsphere/providers.tf [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  description = "Spectro Cloud Project (e.g: Default)"
+  default     = "Default"
+}
+
+provider "spectrocloud" {
+  host         = var.sc_host
+  username     = var.sc_username
+  api_key      = var.sc_api_key
+  # Access via password
+  # password     = var.sc_password
```
Comment (nikchern) [inline examples/resources/spectrocloud_cluster_vsphere/providers.tf:40]: Please remove.

File: examples/resources/spectrocloud_cluster_vsphere/providers.tf [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  description = "Spectro Cloud Username"
+}
+
+variable "sc_api_key" {
+  description = "Spectro API key"
+  sensitive   = true
+  default     = ""
+}
+
+variable "sc_password" {
```
Comment (nikchern) [inline examples/resources/spectrocloud_cluster_vsphere/providers.tf:25]: remove

File: examples/resources/spectrocloud_cluster_vsphere/providers.tf [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    }
+  }
+}
+
+variable "sc_host" {
+  description = "Spectro Cloud Endpoint"
+  default     = "api.spectrocloud.com"
+}
+
+variable "sc_username" {
```
Comment (nikchern) [inline examples/resources/spectrocloud_cluster_vsphere/providers.tf:15]: remove.

**Review by nikchern — CHANGES_REQUESTED** [review]

File: examples/resources/spectrocloud_cluster_vsphere/terraform.templare.tfvars (hunk 4 above)
Comment (SivaanandM) [reply examples/resources/spectrocloud_cluster_vsphere/terraform.templare.tfvars]: removed

File: examples/resources/spectrocloud_cluster_vsphere/providers.tf (hunk 6 above)
Comment (SivaanandM) [reply examples/resources/spectrocloud_cluster_vsphere/providers.tf]: done

File: examples/resources/spectrocloud_cluster_vsphere/providers.tf (hunk 5 above)
Comment (SivaanandM) [reply examples/resources/spectrocloud_cluster_vsphere/providers.tf]: done

File: examples/resources/spectrocloud_cluster_vsphere/providers.tf (hunk 7 above)
Comment (SivaanandM) [reply examples/resources/spectrocloud_cluster_vsphere/providers.tf]: done

**Review by nikchern — APPROVED** [review]

**Pull request merged**